pkg/storage/storage.go

```diff
@@ -45,32 +43,51 @@ func NewStorage() *storage {
 	return &storage{}
 }
 
-// TODO(directxman12): figure out what the right value is for "window" --
-// we don't get the actual window from cAdvisor, so we could just
-// plumb down metric resolution, but that wouldn't be actually correct.
+// Ready returns true if metrics-server's storage has accumulated enough metric
+// points to be able to serve NodeMetrics and PodMetrics.
+func (p *storage) Ready() bool {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+
+	if len(p.prevNodes) == 0 {
+		return false
+	}
+
+	for podIdent := range p.prevPods {
```

Comment:
I think checking nodes is enough for readiness. Technically we could collect metrics from node that doesn't have any pods running, in that case MS would be healthy.

Reply:
good point :+1: